### ISSUE TITLE
Create method quick fix

### ISSUE THREAD

**Issue by MichaelMcDonnell** [rust-analyzer/rust-analyzer#5938]
Would it be possible to add a create method quick fix? In other languages I often write code against an interface. If I need a new method in the interface I just call it before implementing it. I then rely on the IDE's quick fix to add the new method stub for me. For example if I have the following code where the `on_event` method does not exist yet:

```rust
let my_interface = MyInterface::new();
let event: Event = get_event();
my_interface.on_event(event);
```

I would like to have a quick fix to add the `on_event` method to the `MyInterface` impl.

```rust
impl MyInterface {
    fn new() -> Self {...}

    fn on_event(&self, event: Event) {
    }
}
```
The name of the method, type and name of the parameter is straightforward to deduce. It can even be possible to deduce the return type in some cases. The self parameter can also be deduced because it is calling into an impl and not a function in a module. There are things that will not be possible to deduce, like mutable self, but I would be ok with that.

It would be nice if the file with the impl is opened and the cursor placed inside the body of the new method. That would save me from first finding the file and then scrolling to the right place.

UPDATE1: Changed example to snake case.

**Comment by bnjjj** [comment]
I will try to implement this

**Comment by iago-lito** [comment]
There may be several `impl` blocks (perhaps in various files), and no way to pick the "right" one in this case.. What then?
- open a new `impl` block in current file?
- open in existing `impl` block in current file if it exists and is unique?
- open in "closest" impl block back up in the module tree?

**Comment by peddermaster2** [comment]
What about letting the user know, that there is no unique target `impl` block and/or let them choose one?

**Comment by iago-lito** [comment]
Yeah, that would be terrific, but I guess this demands a bit of UI innovation first, does it?
My guess is that the user typically ends up moving the method anyway after it has been written, am I correct? Even within the `impl` block itself. In this case, any existing block would do the trick.. or at least one in the right file?

**Comment by MichaelMcDonnell** [comment]
I keep my `struct` definition and `impl` block in the same file. I never use multiple `impl` blocks for a `struct` definition. I would be perfectly happy if the feature only supports this use case.

I think the best way forward is to start with the simplest case and then iterate as more user feedback is received.

**Comment by MichaelMcDonnell** [comment]
This was implemented in pull request #9804.
